src-tauri/src/main.rs

```diff
@@ -85,66 +82,70 @@ async fn main() {
         .and_then(|s| log::LevelFilter::from_str(s).ok())
         .unwrap_or(log::LevelFilter::Info);
 
-    wooting_macro_backend::MacroBackend::generate_directories();
+    MacroBackend::generate_directories().expect("unable to generate config directories");
 
     let backend = MacroBackend::default();
 
-    info!("Running the macro backend");
-
     #[cfg(any(target_os = "windows", target_os = "linux"))]
-    backend.init().await;
+    // The backend is run only on Windows and Linux, on macOS it won't work.
+    info!("Running the macro backend");
+    if let Err(e) = backend.init().await {
+        eprintln!("Initialization error: {}", e);
+    };
 
+    // Read the options from the config.
     let set_autolaunch: bool = backend.config.read().await.auto_start;
     let set_launch_minimized: bool = backend.config.read().await.minimize_at_launch;
 
-    // Begin the main event loop. This loop cannot run on another thread on MacOS.
+    // Set the window behaviors (add buttons to the right click tray menu).
     let quit = CustomMenuItem::new("quit".to_string(), "Quit");
     let hide_show = CustomMenuItem::new("hide_show".to_string(), "Hide");
-
     let tray_menu = SystemTrayMenu::new()
         .add_item(hide_show)
         .add_native_item(SystemTrayMenuItem::Separator)
         .add_item(quit);
-
     let system_tray = SystemTray::new().with_menu(tray_menu);
 
+    // Initialize the main application
     tauri::Builder::default()
-        // This is where you pass in your commands
+        // State management is initialized
         .manage(backend)
+        // This is where commands shared with frontend are passed
         .invoke_handler(tauri::generate_handler![
             get_macros,
             set_macros,
             get_config,
             set_config,
-            control_grabbing,
-            get_monitor_data
+            control_grabbing
         ])
         .setup(move |app| {
             let app_name = &app.package_info().name;
-            let current_exe = current_exe().unwrap();
+            if let Ok(current_exe) = current_exe() {
```

Comment:
In what situation would current_exe not be Ok? I'm not sure if you really need to be able to safely handle it not being Ok.

Reply:
I think there might be just a filesystem error, permissions issue with the exe location (some weird install location on multiuser systems), things like that. Idea was similar as when disabling autostart, if there is an error, the entire app shouldn't crash.